### PR TITLE
[CP-1033] Add Outbox endpoint to service-desktop

### PR DESCRIPTION
**Pull request by pawel-mudita**
Outbox endpoint will be used to provide notifications
about changes in messages/threads/calls during connection
with Mudita Center. This mechanism will ensure an efficient
data synchronization between Mudita Pure and Mudita Center.

### CODE REVIEW

File: module-services/service-desktop/parser/MessageHandler.cpp [hunk 1]
```diff
@@ -37,12 +38,17 @@ namespace sdesktop::endpoints
...
             LOG_ERROR("No way to handle!");
+            return;
+        }
+        handler->handle(*context);
+        auto ownerService = dynamic_cast<ServiceDesktop *>(OwnerServicePtr);
+        if (ownerService == nullptr) {
+            LOG_ERROR("Downcasting error!");
+            return;
         }
+        ownerService->restartNotificationsClearTimer();
```
Comment (pholat) [inline module-services/service-desktop/parser/MessageHandler.cpp:51]: why not just bind ownerService to the ServiceDesktop? or better: pass dependencies in ctor, in this case: callback to `clear`. or: better: if timer is required by this scope: why not create it here?

**Review by pholat — APPROVED** [review]
seems ok, would be nice if you applied the comment :bow:

File: module-services/service-desktop/parser/MessageHandler.cpp (hunk 1 above)
Comment (pawel-mudita) [reply module-services/service-desktop/parser/MessageHandler.cpp]: Done by passing callback

File: module-services/service-desktop/endpoints/include/endpoints/EndpointType.hpp [hunk 2]
```diff
@@ -21,9 +21,10 @@ namespace sdesktop::endpoints
         calendarEventsPlaceholder,
         developerMode,
         bluetooth,
-        usbSecurity
+        usbSecurity,
+        outbox
     };
 
-    inline constexpr auto lastEndpoint = static_cast<int>(EndpointType::usbSecurity);
+    inline constexpr auto lastEndpoint = static_cast<int>(EndpointType::outbox);
```
Comment (pholat) [inline module-services/service-desktop/endpoints/include/endpoints/EndpointType.hpp:28]: just saying - magic_enum in the repo has the capability to get enum length. I do not know if I would advise using it in the header (as it's heavy on compilation) but it's nice to know :)

**Review by pholat — APPROVED** [review]

**Pull request merged**